telegram/ext/filters.py

```diff
@@ -603,6 +603,66 @@ def filter(self, message: Message) -> bool:
         xml = mime_type('application/xml')
         zip = mime_type('application/zip')
 
+        class file_extension(MessageFilter):
+            """This filter filters documents by their file ending/extension.
+
+            Note:
+                * This Filter only filters by the file ending/extension of the document,
+                    it doesn't check the validity of document.
+                * The user can manipulate the file extension of a document and
+                    send media with wrong types that don't fit to this handler.
+                * Case insensitive by default,
+                    you may change it with the flag ``case_sensitive=True``.
+                * Extension should be passed without leading dot
+                    unless it's a part of the extension.
+                * Pass :obj:`None` to filter files with no extension,
+                    i.e. without a dot in the filename.
+
+            Example:
+                ``Filters.document.file_extension("jpg")``
+                    filters files with extension ``".jpg"``.
+                ``Filters.document.file_extension(".jpg")``
+                    filters files with extension ``"..jpg"``.
+                ``Filters.document.file_extension("Dockerfile", case_sensitive=True)``
+                    filters files with extension ``".Dockerfile"`` minding the case.
+            """
+
+            def __init__(self, file_extension: Optional[str], case_sensitive: bool = False):
+                """Initialize the extension you want to filter.
+
+                Args:
+                    file_extension (:obj:`str` | :obj:`None`):
+                        media file extension you want to filter.
+                    case_sensitive (:obj:bool, optional):
+                        pass :obj:`True` to make the filter case sensitive.
+                        Default: :obj:`False`.
+                """
+                self.is_case_sensitive = case_sensitive
+                if file_extension is None:
+                    self.file_extension = None
+                    self.name = "Filters.document.file_extension(None)"
+                elif case_sensitive:
+                    self.file_extension = f".{file_extension}"
+                    self.name = (
+                        f"Filters.document.file_extension({file_extension!r},"
```

Comment:
sorry what is the !r for?

Reply:
it adds quotes to the strings. Also make `None` distinguishable from `"None"`. Also asked that :D

Reply:
quick google didnt show that to me :(

Reply:
uses `repr()` instead of `str()`